### ISSUE TITLE
DIY Tracker on LLCC68 doesn't start

### ISSUE THREAD

**Issue by MajakOwO** [richonguzman/LoRa_APRS_Tracker#156]
Hi,
I recently built a diy tracker on esp32 and Ebyte E220 400M30S (LLCC68)
The problem is that it doesn't start, I keep getting this error:
Starting LoRa failed! State: -2
Everything is wired right
What is the possible solution?

**Comment by richonguzman** [comment]
What are your Lora settings? Sf12?

**Comment by MajakOwO** [comment]
> What are your Lora settings? Sf12?

![obraz](https://github.com/user-attachments/assets/0cd6a1d5-c7f1-4582-89c6-8f1d5443e624)
SF9

**Comment by richonguzman** [comment]
> > What are your Lora settings? Sf12?
> 
>  SF9

power is on 18 (valid power units are 17, 20, 22... for your board max is 20). change it!

**Comment by MajakOwO** [comment]
I changed it to 17 and it still doesn't start

**Comment by richonguzman** [comment]
the [RADIOLIB_ERR_CHIP_NOT_FOUND]   (-2) you experience is because it can recognize the Lora Module

please check wiring / pins

**Comment by MajakOwO** [comment]
Wiring is good and it still doesn't start

**Comment by richonguzman** [comment]
did you solve this?

**Comment by MajakOwO** [comment]
No, maybe Lora module is dead?

**Comment by richonguzman** [comment]
could you test it with other firmwares?

**Comment by MajakOwO** [comment]
What other firmwares are there?

**Comment by richonguzman** [comment]
> What other firmwares are there?

is there a meshtastic one for this ESP32 + LLCC68 board?

**Comment by MajakOwO** [comment]
> > What other firmwares are there?
> 
> is there a meshtastic one for this ESP32 + LLCC68 board?

Yes, meshtastic starts on LLCC68 (RA-02SC) but not on E220 400M30S which is also on LLCC68 chip 
![Zrzut ekranu 2024-10-04 150933](https://github.com/user-attachments/assets/a4011f37-e4e9-4623-b221-7856d1cc2ad5)

**Comment by richonguzman** [comment]
I received confirmation of another Ham OP which used ESP32 DIY with LLCC68 board with success

So I would suggest to double check the connections, the esp32 boards and finally see if E220 boards is working

**Comment by richonguzman** [comment]
did you try latest update of tracker?

**Comment by harifhaqim** [comment]
tried with RA-01SC(LLCC68), removed the 1W definition but still...same issue with OP

**Comment by richonguzman** [comment]
> tried with RA-01SC(LLCC68), removed the 1W definition but still...same issue with OP

1W definition is needed!


I heard lots of successful stories about using this chip, but! is your board the 1W version or not? they have different pins!

**Comment by harifhaqim** [comment]
not a 1W chip, it's RA-01SC with LLCC68...if it does need 1W version, the original OP would have been succeeded..and both 1W and 0.1W version have the same library..the LoRa initialization failed from the beginning resulting in -2 error code

![Image](https://github.com/user-attachments/assets/f763f0be-b01e-4767-9157-23e82dff0db9)

**Comment by richonguzman** [comment]
so not found...  maybe have to check on the pins definition

**Comment by harifhaqim** [comment]
i have also made a remap

DIO1 -> 4
BUSY -> 2
RXEN -> 1(TX0)
TXEN -> 3(RX0)

already appended to the code, check continuity and it's fine, the RadioLib said it might be NSS(CS) wrong pin but I've tested it and using the default GPIO..the other Pin using the standard pinout

**Comment by richonguzman** [comment]
first, what are the pins used for this lora unit?

**Comment by harifhaqim** [comment]
#define RADIO_SCLK_PIN      18
    #define RADIO_MISO_PIN      19
    #define RADIO_MOSI_PIN      23
    #define RADIO_CS_PIN        5
    #define RADIO_RST_PIN       27
    #define RADIO_DIO1_PIN      12
    #define RADIO_BUSY_PIN      14
    #define RADIO_RXEN          32
    #define RADIO_TXEN          25

this is the stock pinout not including the remap.

**Comment by richonguzman** [comment]
I know but can you link the product data sheet to see if we really need that much pins defined or maybe check if there is a new different needed

**Comment by harifhaqim** [comment]
this is the module for Ra-01SC I'm using 

![Image](https://github.com/user-attachments/assets/c72eed2d-cc26-4a12-ba4e-d7c9980d037f)

**Comment by harifhaqim** [comment]
> I know but can you link the product data sheet to see if we really need that much pins defined or maybe check if there is a new different needed

[ra-01sc_specification.pdf](https://github.com/user-attachments/files/18814931/ra-01sc_specification.pdf)

**Comment by richonguzman** [comment]
please the data sheet link


I already thing there should be no TxEN and RxEN pins for your board

**Comment by richonguzman** [comment]
now I've seen the data sheet

you need to remap all of them:

#define RADIO_SCLK_PIN      ?
#define RADIO_MISO_PIN      ?
#define RADIO_MOSI_PIN      ?
#define RADIO_CS_PIN        ?
#define RADIO_RST_PIN       ?
#define RADIO_DIO1_PIN     ?
#define RADIO_BUSY_PIN      ?
#define RADIO_RXEN          ?
#define RADIO_TXEN          ?

you had them defined before, but is this wired as you wrote?

**Comment by harifhaqim** [comment]
i used the stock pinout with some of them remapped, so this is my current pinout

#define RADIO_SCLK_PIN      18
#define RADIO_MISO_PIN      19
#define RADIO_MOSI_PIN      23
#define RADIO_CS_PIN        5
#define RADIO_RST_PIN       27

The Remapped Pins
#define RADIO_DIO1_PIN      4
#define RADIO_BUSY_PIN      2
#define RADIO_RXEN          1
#define RADIO_TXEN          3

**Comment by richonguzman** [comment]
are this pins really available on your esp32 ???

> i used the stock pinout with some of them remapped, so this is my current pinout
> 
> #define RADIO_SCLK_PIN 18 #define RADIO_MISO_PIN 19 #define RADIO_MOSI_PIN 23 #define RADIO_CS_PIN 5 #define RADIO_RST_PIN 27
> 
> The Remapped Pins #define RADIO_DIO1_PIN 4 #define RADIO_BUSY_PIN 2 #define RADIO_RXEN 1 #define RADIO_TXEN 3

**Comment by harifhaqim** [comment]
I'm using this ESP32 module 

![Image](https://github.com/user-attachments/assets/f16efa27-682e-4a62-b773-41e3e5cef8c0)

**Comment by richonguzman** [comment]
GPIO + number are the available pins for the ESP32

test more pins like 4, 2, 15 and 14 for free pins as sometimes this pins are used when USB is needed

**Comment by harifhaqim** [comment]
so the TXEN and RXEN are the wrong pin for me?

**Comment by richonguzman** [comment]
> so the TXEN and RXEN are the wrong pin for me?

I would test 12,13,14 for them and test

**Comment by harifhaqim** [comment]
> > so the TXEN and RXEN are the wrong pin for me?
> 
> I would test 12,13,14 for them and test

I have remapped the pin to GPIO 26 for RXEN and GPIO 14 for TXEN....still -2 error

**Comment by richonguzman** [comment]
I don't know what else to do with this board. It seems you've tested all the needed things to make it work
